lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,291 @@
+        vm_name: "{{ vm_name }}"
+        name: "{{ vm-name }}-image"
+        location: "{{ location }}"
+        state: present
+
+    - name: Delete image
+      azure_rm_image:
+        subscription_id: "{{ subscription_id }}"
+        resource_group_name: "{{ resource_group_name }}"
+        client_id: "{{ secrets.client_id }}"
+        tenant_id: "{{ tenant_id }}"
+        client_secret: "{{ secrets.client_secret }}"
+        name: "{{ vm-name }}-image"
+        location: "{{ location }}"
+        state: absent
+'''
+
+RETURN = '''
+location:
+    description: The location where the image is created
+    returned: on success
+    type: string
+name:
+    description: The name of the new image
+    returned: on success
+    type: string
+resource_group:
+    description: The resource group where the image is created
+    returned: on success
+    type: string
+status:
+    description: The result of the operation
+    returned: always
+    type: string
+'''
+
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.compute.models import Image
+    from azure.mgmt.compute.models import SubResource
+# from azure.mgmt.compute.models import VirtualMachineCaptureParameters
+except:
+    pass
+
+
+class AzureRMImage(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            vm_name=dict(
+                type='str',
+                required=False
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        required_if = [
+            ('state', 'present', ['vm_name'])
+        ]
+
+        self.resource_group = None
+        self.vm_name = None
+        self.name = None
+        self.state = None
+        self.location = None
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_images=None)
+        )
+
+        super(AzureRMImage, self).__init__(
+            derived_arg_spec=self.module_arg_spec,
+            required_if=required_if,
+            supports_check_mode=True,
+            supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
```

Comment:
Python 2.6 doesn't support `{}` in format, you need to set the index number like `{0}` and increment for each subsequent block.